### PR TITLE
Refact: set defaults, separate bin files, optionally load dotenv

### PR DESCRIPTION
**Pull request by keenahn**
1. Sets some defaults so that it works without adding anything to package.json
2. Optionally loads dotenv if it is present
3. Separates out the bin files so that nothing is run on `require` and so that you can use the methods programmatically
4. Adds `assert-release-note` and `push-release-note` to its own `package.json` for meta

### CODE REVIEW

File: src/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  filter(overEvery([
+    startsWith('+'),
+    negate(startsWith('++')),
+  ])),
+  map(trimCharsStart('+')),
+  join('\n'),
+  trim,
+)
+
+const getLastChangelogAdditions = ({ changelogFile = 'CHANGELOG.md', commitHash = 'master' } = {}) => {
```
Comment (keenahn) [inline src/helpers/index.js]: use constants

File: package.json [hunk 2]
```diff
@@ -39,23 +36,26 @@
...
   "repository": "git@github.com:invisible-tech/release-note.git",
   "scripts": {
     "builddoc": "./node_modules/.bin/jsdoc -c ./.jsdoc.json",
     "coverage": "nyc report --reporter=lcov --reporter=html --reporter=text-summary --reporter=text",
     "lint": "eslint --cache -f node_modules/eslint-formatter-pretty --ignore-path .gitignore .",
     "start": "node src",
-    "pretest": "npm run lint",
-    "test": "exit 0",
-    "posttest": "assert-version-bump && nsp check"
+    "prepublish": "nsp check",
```
Comment (keenahn) [inline package.json]: prepublish being deprecated

File: bin/assert-release-note [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const finder = require('find-package-json')
+const { get } = require('lodash/fp')
+
+const assertReleaseNote = require('../src/assertReleaseNote')
+
+const pkg = finder().next().value || {}
+
+try {
+  assertReleaseNote({ changelogFile: get('releaseNote.changelogFile')(pkg) || 'CHANGELOG.md' })
```
Comment (keenahn) [inline bin/assert-release-note]: can set the default bot name from the package name

**Comment by rodrigo4244** [comment]
There is a difference between using `execa` and `cross-spawn`. On the latter you need to use `{ encoding: 'utf8' }` because it returns a buffer exactly as `child_process`, the former uses `utf8` encoding by default.

After fixing this, feel free to merge!

**Comment by keenahn** [comment]
https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options 

Child process spawn defaults to encoding utf8 :)

**Comment by keenahn** [comment]
owait, you were right

**Pull request merged**